### PR TITLE
Expands perma, adds license plate presses

### PR DESCRIPTION
**Pull request by bruhlookatthisdood**
# Document the changes in your pull request

I gave perma an expansion
I also gave it license plates
Cope?
It dosent touch the cells, shouldn't interfere with PR #13893
Box station has a disposal unit that goes from Warden to the prisoner's cell... for licensep lates.
image
image
image

# Wiki Documentation

New picture

# Changelog

:cl:  Spookydonut and bruhlookatthisdood
rscadd: Perma has been given a makeover
rscadd: Perma has been given license plate presses to give them something to work on in prison.
/:cl:

### CODE REVIEW

**Comment by Cartlord** [comment]
your changelog has a typo
shame

**Comment by KillerOrcaCora** [comment]
Based PR

**Comment by adamsong** [comment]
I like the part where perma prisoners are given access to the warden's office

File: code/game/machinery/prisonlabor.dm [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/obj/machinery/plate_press/attackby(obj/item/I, mob/living/user, params)
+	if(!is_operational())
+		to_chat(user, "<span class='warning'>[src] has to be on to do this!</span>")
+		return FALSE
+	if(pressing)
+		to_chat(user, "<span class='warning'>[src] already has a plate in it!</span>")
+		return FALSE
+	if(istype(I, /obj/item/stack/license_plates/empty))
+		var/obj/item/stack/license_plates/empty/plate = I
+		plate.use(1)
```
Comment (jjpark-kb) [inline code/game/machinery/prisonlabor.dm:36]: maybe put this as an if statement. "If you cannot use, early return."

File: code/game/machinery/prisonlabor.dm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		current_plate = new plate.type(src, 1) //Spawn a new single sheet in the machine
+		update_icon()
+	else
+		return ..()
+
+/obj/machinery/plate_press/attack_hand(mob/living/user)
+	. = ..()
+	if(!pressing && current_plate)
+		work_press(user)
```
Comment (jjpark-kb) [inline code/game/machinery/prisonlabor.dm]: early return might be good here.
```suggestion
	if(pressing || !current_plate)
		return
	work_press(user)
```

**Comment by bruhlookatthisdood** [comment]
> I like the part where perma prisoners are given access to the warden's office

how? its a outlet.

**Comment by maxion12345** [comment]
Expansion seems a bit too large to justify another a few machines. This also ruins the whole "openness" of Perma that really solidified the whole "this is jail, the cameras watch your every move, there is no privacy outside of your cell and the bathroom".

Also disposal pipes through walls bad

File: code/game/machinery/prisonlabor.dm (hunk 1 above)
Comment (bruhlookatthisdood) [reply code/game/machinery/prisonlabor.dm:36]: dont see why it would be neccesary.

**Comment by JamieD1** [comment]
Comments in DMs

File: code/game/machinery/prisonlabor.dm [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	. = ..()
+
+/obj/machinery/plate_press/attackby(obj/item/I, mob/living/user, params)
+	if(!is_operational())
+		to_chat(user, "<span class='warning'>[src] has to be on to do this!</span>")
+		return FALSE
+	if(pressing)
+		to_chat(user, "<span class='warning'>[src] already has a plate in it!</span>")
+		return FALSE
+	if(istype(I, /obj/item/stack/license_plates/empty))
```
Comment (Bibby0110) [inline code/game/machinery/prisonlabor.dm]:
```suggestion
	if(!istype(I, /obj/item/stack/license_plates/empty))
		return ..()
```

File: code/game/machinery/prisonlabor.dm [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		return ..()
+
+/obj/machinery/plate_press/attack_hand(mob/living/user)
+	. = ..()
+	if(!pressing && current_plate)
+		work_press(user)
+
+///This proc attempts to create a plate. User cannot move during this process.
+/obj/machinery/plate_press/proc/work_press(mob/living/user)
+
```
Comment (Bibby0110) [inline code/game/machinery/prisonlabor.dm]:
```suggestion
```

File: code/modules/cargo/packs.dm [hunk 5]
```diff
@@ -1026,6 +1026,13 @@
 	contains = list(/obj/item/stack/sheet/cardboard/fifty)
 	crate_name = "cardboard sheets crate"
 
+/datum/supply_pack/materials/license50
+	name = "50 Empty License Plates"
+	desc = "'Put those prisoners to work' - Jedi Yoda probably"
+	cost = 1000  // 50 * 25 + 700 - 1000 = 950 credits profit
```
Comment (Bibby0110) [inline code/modules/cargo/packs.dm]: +700?

**Review by Bibby0110 — DISMISSED** [review]

**Review by nesisknesis — APPROVED** [review]

**Comment by bruhlookatthisdood** [comment]
only thing im worried about right now is the warden throwing things in the license plate disposal chute, that are not license plates.

the prisoners must mald.

**Review by JamieD1 — APPROVED** [review]

**Comment by Cartlord** [comment]
But what do you DO with license plates?

**Comment by bruhlookatthisdood** [comment]
> But what do you DO with license plates?

sell them to cargo

**Comment by ghost** [comment]
Actually it conflicts with https://github.com/yogstation13/Yogstation/pull/13893

**Comment by bruhlookatthisdood** [comment]
> Actually it conflicts with #13893

uhoh

**Pull request merged**